base/rational.jl

```diff
@@ -98,9 +98,26 @@ function //(x::Rational, y::Rational)
 end
 
 //(x::Complex, y::Real) = complex(real(x)//y, imag(x)//y)
-//(x::Number, y::Complex) = x*conj(y)//abs2(y)
-
-
+function //(x::Number, y::Complex)
+    if((x//abs2(y))==0//1 || (x//abs2(y))==1//0)
+        return (x//abs2(y))
+    end
+    return (x//abs2(y))*conj(y)
+end
+function //(x::Number, y::Complex{<:Integer})
+    if isinf(real(y)) || isinf(imag(y))
+        return 0//1
+    end
+    real_y = real(y)
+    imag_y = imag(y)
+    denom = Int32(abs(real_y))^2 + Int32(abs(imag_y))^2
```

Comment:
Why `Int32` here?

Reply:
I added that to account for any overflow, but int16 would work perfectly fine as well!

Reply:
Hey, I’ve implemented scaling to prevent any overflow in calculations (removing any need for INT32. Could you take a look at my approach